File: Plan Graph (sgware)/src/edu/uno/ai/planning/pg/PlanGraphPlanner.java
package edu.uno.ai.planning.pg;

import edu.uno.ai.planning.Planner;
import edu.uno.ai.planning.Problem;
import edu.uno.ai.planning.Search;

public abstract class PlanGraphPlanner extends Planner<PlanGraphSearch> {

	public PlanGraphPlanner(String name) {
		super(name);
	}
	
	@Override
	protected PlanGraphSearch makeSearch(Problem problem) {
		return new PlanGraphSearch(this, problem);
	}
	
	protected abstract Search makeSearch(PlanGraph graph);
}

File: UnitySokoban/Assets/Scripts/Planning/PlanGraphSGW/PlanGraphPlanner.cs
using Planning;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PlanGraphSGW
{
    public abstract class PlanGraphPlanner : Planner<PlanGraphSearch>
    {
        public PlanGraphPlanner(String name)
            : base(name)
        {
        }

        public override PlanGraphSearch makeSearch(Problem problem)
        {
            return new PlanGraphSearch(this, problem);
        }

        public abstract Search makeSearch(PlanGraph graph);
    }
}
